mmdet/datasets/pipelines/test_time_aug.py

```diff
@@ -29,8 +29,7 @@ def __init__(self,
                  flip=False,
                  flip_direction='horizontal'):
         self.transforms = Compose(transforms)
-        assert img_scale or scale_factor
-        assert None in [img_scale, scale_factor]
+        assert (img_scale is None) ^ (scale_factor is None)
```

Comment:
Is `scale_factor` tested and observed gains in the corresponding model? As you can find in line 61, the result is shallow copied. If the scale factor is [0.8, 1, 1.2], the behavior might not be what we expected, i.e., the shape of the image (assumed to have size 512) might become `512*0.8`, `512*0.8*1`, and `512*0.8*1*1.2`, rather than `512*0.8`, `512*1`, and `512*1.2`.

Reply:
This code works as expected in my env (python3.6). Due to the shallow copy, `id(results['img']) == id(_results['img'])` , but `mmcv.imrescale` will return a new img with different id in `Resize` op, so I think the shallow copy won't lead to bad effect. I also added an unittest for this case.